### ISSUE TITLE
Refer to this.someProperty in a function() {}

### ISSUE THREAD

**Issue by jensravn** [Microsoft/TypeScript#14543]
**TypeScript Version:**  2.1.1 / nightly (2.2.0-dev.201xxxxx)

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
class Example {

    public message: string = "Hello";

    public jsFunc() {   
        setTimeout(function() { alert(this.message); }, 0);
    }
}

let example = new Example();

example.jsFunc();
```

**Expected behavior:**
I would expect it to alert `"Hello"`.
It does so if I write `() => { alert(this.message); }` instead of `function() { alert(this.message); }`.
I can see that the difference in the compiled javascript is that the lambda syntax `() => {}` assigns `var _this = this;` 
I would expect the javascript syntax `function() { }` to be compiled to the same code as the lambda
or show me a warning in TypeScript that `this.message` is undefined.

**Actual behavior:**
TypeScript does not warn that `this.message` is undefined.
When running the code it alerts `"Undefined"`.

**Comment by RyanCavanaugh** [comment]
You can run with `--noImplicitThis` to detect these mistakes.

**Comment by jensravn** [comment]
Oh, that was embarrassing. I should have seen that. Thank you very much :)
